framework/src/task_manager/task_manager_get_new_broadcast_msg.c

```diff
@@ -0,0 +1,63 @@
+
+/****************************************************************************
+ *
+ * Copyright 2018 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <string.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <task_manager/task_manager.h>
+#include "task_manager_internal.h"
+
+/****************************************************************************
+ * task_manager_get_new_broadcast_msg
+ ****************************************************************************/
+int task_manager_get_new_broadcast_msg(int timeout)
+{
+	int status;
+	tm_request_t request_msg;
+	tm_response_t response_msg;
+
+	if (timeout < TM_RESPONSE_WAIT_INF) {
+		return TM_INVALID_PARAM;
+	}
+
+	memset(&request_msg, 0, sizeof(tm_request_t));
+	request_msg.cmd = TASKMGRCMD_GET_NEW_BROADCAST_MSG;
+	request_msg.caller_pid = getpid();
+	request_msg.timeout = timeout;
+	
+	if (timeout != TM_NO_RESPONSE) {
```

Comment:
You had better put his code inside of task_manager_send_request, then you can reduce many duplicated codes.
```
int taskmgr_send_request(tm_request_t *request_msg) 
{
   ...
   if (request_msg.timeout != TM_NO_RESPONSE) {
   ...
   } 
}
```

Reply:
There are some other operations for `if (timeout != TM_NO_RESPONSE)` condition,  in other functions of task manager. Therefore, it can not be merge to the taskmgr_send_request() function.